apps/web-giddh/src/app/activity-logs/activity-logs.component.ts

```diff
@@ -298,6 +316,87 @@ export class ActivityLogsComponent implements OnInit, OnDestroy {
     }
 
 
+
+    /**
+     * To check is entry expanded
+     *
+     * @param {*} entry Transaction object
+     * @memberof ActivityLogsComponent
+     */
+    public getHistory(event: any, row: any): void {
+        event.stopPropagation();
+        event.preventDefault();
+        row.isExpanded = !row.isExpanded;
+        if (!row.hasHistory) {
+            let activityObj = { entityId: row.entityId, entity: row.entity, count: 100 };
+            row.hasHistory = true;
+            this.activityService.getActivityLogs(activityObj).pipe(takeUntil(this.destroyed$)).subscribe((response) => {
+                this.isLoading = false;
+                if (response && response.status === 'success') {
+                    response.body?.results?.forEach((result, index) => {
+                        if (result) {
+                            result.index = index;
+                            result.timeonly = moment(result.time, GIDDH_DATE_FORMAT + " HH:mm:ss").format("HH:mm:ss");
+                            result.time = moment(result.time, GIDDH_DATE_FORMAT + " HH:mm:ss").format(GIDDH_DATE_FORMAT);
+                        }
+                    });
+                    row.history = response.body.results;
+                } else {
+                    row.history = [];
+                }
+                this.changeDetection.detectChanges();
+            });
+        }
+    }
+
+    /**
+     * This function will use for selected items
+     *
+     * @param {MatCheckboxChange} event
+     * @param {*} row2
+     * @param {*} details
+     * @memberof ActivityLogsComponent
+     */
+    public selectedItems(event: MatCheckboxChange, row2: any, details: any): void {
+        if (!row2.selectedItems) {
+            row2.selectedItems = [];
+        }
+        if (event.checked) {
+            details.isChecked = true;
+            this.changeDetection.detectChanges();
```

Comment:
Consider calling detectChanges at the end of the method as it is expensive operation